packages/apollo-server-hapi/src/hapiApollo.ts

```diff
@@ -34,56 +39,100 @@ const graphqlHapi: IPlugin = {
     if (!options || !options.graphqlOptions) {
       throw new Error('Apollo Server requires options.');
     }
-    server.route({
-      method: ['GET', 'POST'],
-      path: options.path || '/graphql',
-      vhost: options.vhost || undefined,
-      options: options.route || {},
-      handler: async (request, h) => {
```

Comment:
This handler function's body – with some `query` assignment changes – is copied to the `handler` function.